### PR TITLE
Update changelog and version for 2.6rc3

### PR DESCRIPTION
**Pull request by sloosvel**
## Description

This PR updates the changelog and the Citation file for the release of v2.6.0rc3

Link to documentation: https://esmvaltool--1646.org.readthedocs.build/projects/ESMValCore/en/1646/changelog.html#v2-6-0

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

~- [ ] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do~

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

~- [ ] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)~
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
~- [ ] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added~
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
~- [ ] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly~
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -194,5 +194,5 @@ license: "Apache-2.0"
 message: "If you use this software, please cite it using these metadata."
 repository-code: "https://github.com/ESMValGroup/ESMValCore/"
 title: ESMValCore
-version: "v2.6.0rc2"
+version: "v2.6.0rc3"
```
Comment (schlunma) [inline CITATION.cff:197]: Please make sure to also change `date-released` (GitHub won't let me do that here)

File: doc/changelog.rst [hunk 2]
```diff
@@ -57,14 +61,20 @@ Documentation
...
 -  Remove support for Mistral in `config-user.yml` (`#1620 <https://github.com/ESMValGroup/ESMValCore/pull/1620>`__) `Rémi Kazeroni <https://github.com/remi-kazeroni>`__
+-  Add changelog for v2.6.0rc1 (`#1633 <https://github.com/ESMValGroup/ESMValCore/pull/1633>`__) `sloosvel <https://github.com/sloosvel>`__
+-  Add a note on transferring permissions to the release manager (`#1645 <https://github.com/ESMValGroup/ESMValCore/pull/1645>`__) `Bouwe Andela <https://github.com/bouweandela>`__
 
 Improvements
 ~~~~~~~~~~~~
 
 -  Removed trailing whitespace in custom CMOR tables (`#1564 <https://github.com/ESMValGroup/ESMValCore/pull/1564>`__) `Manuel Schlund <https://github.com/schlunma>`__
 -  Try searching multiple ESGF index nodes (`#1561 <https://github.com/ESMValGroup/ESMValCore/pull/1561>`__) `Bouwe Andela <https://github.com/bouweandela>`__
+-  Add general `extract_coordinate_points` preprocessor (`#1581 <https://github.com/ESMValGroup/ESMValCore/pull/1581>`__) `sloosvel <https://github.com/sloosvel>`__
```
Comment (schlunma) [inline doc/changelog.rst:72]: Why has this moved from "Preprocessor" to "Improvements"? I think it fits better in "Preprocessor"

**Review by schlunma — COMMENTED** [review]
Looks good! Just 2 small comments.

Please also make sure to tick the boxes above :grin:

File: doc/changelog.rst (hunk 2 above)
Comment (schlunma) [reply doc/changelog.rst:72]: (Same for "Add preprocessor `accumulate_coordinate`")

Comment (sloosvel) [reply doc/changelog.rst]: Yes it fits better in preproc, I guess I touched something in `draft_release_notes` that should not have been changed

File: doc/changelog.rst [hunk 3]
```diff
@@ -57,14 +61,20 @@ Documentation
...
 -  Add CMIP6 `amoc` derivation case and add a test (`#1577 <https://github.com/ESMValGroup/ESMValCore/pull/1577>`__) `Valeriu Predoi <https://github.com/valeriupredoi>`__
 -  Added EMAC CMORizer (`#1554 <https://github.com/ESMValGroup/ESMValCore/pull/1554>`__) `Manuel Schlund <https://github.com/schlunma>`__
+-  Add preprocessor `accumulate_coordinate` (`#1281 <https://github.com/ESMValGroup/ESMValCore/pull/1281>`__) `Javier Vegas-Regidor <https://github.com/jvegreg>`__
+-  Add tests and docs (`#1614 <https://github.com/ESMValGroup/ESMValCore/pull/1614>`__) `Zeb Nicholls <https://github.com/znicholls>`__
```
Comment (schlunma) [inline doc/changelog.rst:74]: This can be removed I guess, it's a PR into a feature branch (not `main`)

Comment (sloosvel) [reply doc/changelog.rst]: Ok, I have also added #1644 since it got merged just now

**Review by schlunma — APPROVED** [review]
Nice!

**Pull request merged**